src/peft/utils/save_and_load.py

```diff
@@ -72,6 +76,15 @@ def get_peft_model_state_dict(model, state_dict=None, adapter_name="default", un
                 config.rank_pattern = rank_pattern
                 to_return = model.resize_state_dict_by_rank_pattern(rank_pattern, to_return, adapter_name)
 
+        # save the corresponding base embedding layers when loras are applied to the embedding layers
+        # support from version >= 0.6.2
+        if base_model_layers_to_save is not None:
+            if not isinstance(base_model_layers_to_save, list):
+                raise ValueError(f"{base_model_layers_to_save} should be a list of strings")
+            for k, v in state_dict.items():
+                if any(f"{module_name}.base_layer" in k for module_name in base_model_layers_to_save):
```

Comment:
I think this will work for now but could fail once we add support for mixed adapters. When we have a mixed adapter, we can have, for example, a `LoraLayer` whose `base_layer` is a `LoHaLayer` whose `base_layer` is `nn.Linear`. Not 100% sure what the correct solution is, maybe check if the `base_layer` is _not_ a `BaseTunerLayer`?